File: coldcall/MenuViewController.m
//
//  MenuViewController.m
//  RevealControllerStoryboardExample
//
//  Created by Nick Hodapp on 1/9/13.
//  Copyright (c) 2013 CoDeveloper. All rights reserved.
//
#import "SWRevealViewController.h"
#import "MenuViewController.h"
#import "coldcall-Swift.h"

@implementation SWUITableViewCell3
@end

@implementation MenuViewController3

- (void) prepareForSegueB: (UIStoryboardSegue *) segue sender: (id) sender
{
    // configure the destination view controller:
    if ( [segue.destinationViewController isKindOfClass: [ColorViewController class]] &&
        [sender isKindOfClass:[UITableViewCell class]] )
    {
        UILabel* c = [(SWUITableViewCell3 *)sender label];
        ColorViewController* cvc = segue.destinationViewController;
        
        cvc.color = c.textColor;
        cvc.text = c.text;
    }

    // configure the segue.
    if ( [segue isKindOfClass: [SWRevealViewControllerSegue class]] )
    {
        SWRevealViewControllerSegue* rvcs = (SWRevealViewControllerSegue*) segue;
        
        SWRevealViewController* rvc = self.revealViewController;
        NSAssert( rvc != nil, @"oops! must have a revealViewController" );
        
        NSAssert( [rvc.frontViewController isKindOfClass: [UINavigationController class]], @"oops!  for this segue we want a permanent navigation controller in the front!" );

        rvcs.performBlock = ^(SWRevealViewControllerSegue* rvc_segue, UIViewController* svc, UIViewController* dvc)
        {
            UINavigationController* nc = [[UINavigationController alloc] initWithRootViewController:dvc];
            [rvc pushFrontViewController:nc animated:YES];
        };
    }
}


- (void) prepareForSegue: (UIStoryboardSegue *) segue sender: (id) sender
{
    // configure the destination view controller:
    if ( [sender isKindOfClass:[UITableViewCell class]] )
    {
        UILabel* c = [(SWUITableViewCell3 *)sender label];
        UINavigationController *navController = segue.destinationViewController;
        ColorViewController* cvc = [navController childViewControllers].firstObject;
        if ( [cvc isKindOfClass:[ColorViewController class]] )
        {
            cvc.color = c.textColor;
            cvc.text = c.text;
        }
        if ( [cvc isKindOfClass:[NewCCTableViewController class]] )
        {
           // insert seque information for NewViewController Class
        }
    }
}


#pragma mark - Table view data source

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
{
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return 4;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    static NSString *CellIdentifier = @"Cell";

    switch ( indexPath.row )
    {
        case 0:
            CellIdentifier = @"coldcalls";
            break;
            
        case 1:
            CellIdentifier = @"contacts";
            break;

        case 2:
            CellIdentifier = @"competition";
            break;
        case 3:
            CellIdentifier = @"settings";
    }

    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier: CellIdentifier forIndexPath: indexPath];
 
    return cell;
}

@end

File: MenuViewController.swift
//
//  MenuTableViewController.swift
//  coldcall
//
//  Created by Jason Crump on 7/22/14.
//  Copyright (c) 2014 Jason Crump. All rights reserved.
//

import UIKit

class SWUITableViewCell : UITableViewCell {
    @IBOutlet var label : UILabel!
}

class MenuViewController: UITableViewController, UserSessionControllerDelegate {
    
    override func viewDidLoad() {
        super.viewDidLoad()
        var userSession = UserSessionController.sharedInstance
        userSession.delegate = self
        
    }
    
    override func prepareForSegue(segue: UIStoryboardSegue!, sender: AnyObject!) {
        if sender.isKindOfClass(UITableViewCell)
        {
        }
    }
    
    func prepareForSegueB(segue: UIStoryboardSegue!, sender: AnyObject!) {
        if(segue.isKindOfClass(SWRevealViewControllerSegue))
        {
            var rvcs : SWRevealViewControllerSegue = segue as SWRevealViewControllerSegue
            var rvc : SWRevealViewController = self.revealViewController()
            
            rvcs.performBlock = {(rvc_segue, svc, dvc) in
                var nc:UINavigationController = dvc as UINavigationController
                rvc.pushFrontViewController(nc, animated: true)
            }
        }
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }

    // MARK: - Table view data source
    
    func userSessionChanged() {
        println("userSessionChanged")
        tableView.reloadData()
    }

    override func numberOfSectionsInTableView(tableView: UITableView!) -> Int {
        return 1
    }
    
    override func tableView(tableView: UITableView!, heightForRowAtIndexPath indexPath: NSIndexPath!) -> CGFloat {
        println(indexPath.row)
        var rowHeight : CGFloat = 0
        if indexPath.row == 0 {
            rowHeight = 65.0
        } else {
            rowHeight = 50.0
        }
        return rowHeight
    }

    override func tableView(tableView: UITableView!, numberOfRowsInSection section: Int) -> Int {
        return 6
    }
    
    override func tableView(tableView: UITableView!, cellForRowAtIndexPath indexPath: NSIndexPath!) -> UITableViewCell! {
        let cellIdentifiers = ["title", "coldcalls", "prospects", "contacts", "map", "settings"]
        var cellIdentifier = "Cell"
        cellIdentifier = cellIdentifiers[indexPath.row]
        
        var cell : UITableViewCell? = tableView.dequeueReusableCellWithIdentifier(cellIdentifier, forIndexPath: indexPath) as? UITableViewCell
        if cellIdentifier == "title" {
            for sub in cell!.subviews {
                sub.removeFromSuperview()
            }
            let userSession = UserSessionController.sharedInstance
            var imageView = UIImageView(frame: CGRectMake(12,1,20,30))
            imageView.image = userSession.getUserImage()
            imageView.sizeToFit()
            imageView.layer.cornerRadius = imageView.frame.size.width / 2
            imageView.clipsToBounds = true
            imageView.center.y = (65.0 / 2)
            imageView.layer.borderWidth = 3.0
            imageView.layer.borderColor = UIColor.darkGrayColor().CGColor

            var nameLabel = UILabel(frame: CGRectMake(72,12,200,20))
            nameLabel.text = userSession.userName().uppercaseString
            nameLabel.textColor = UIColor.darkGrayColor()
            nameLabel.font = UIFont(name: "HelveticaNeue-Thin", size: 20)
            nameLabel.center.y = imageView.center.y
            cell?.addSubview(imageView)
            cell?.addSubview(nameLabel)
        }
        return cell
    }

}
